models/Asset/Video/ImageThumbnail.php

```diff
@@ -131,19 +132,33 @@ public function generate(bool $deferredAllowed = true): void
 
                 if (!$storage->fileExists($cacheFilePath)) {
                     $lock = Pimcore::getContainer()->get(LockFactory::class)->createLock($cacheFilePath);
-                    $lock->acquire(true);
-
-                    // after we got the lock, check again if the image exists in the meantime - if not - generate it
-                    if (!$storage->fileExists($cacheFilePath)) {
+                    if ($lock->acquire()) {
```

Comment:
not sure, but there's something in this change that doesn't feel right there
https://symfony.com/doc/current/components/lock.html#blocking-locks

Is there any drawbacks with this change?

Reply:
I wanted to unify it with "models/Asset/Document/ImageThumbnail.php". But I can change it back to Blocking Lock so the behavior doesn't change.

Reply:
@kingjia90 Done.